CorsixTH/Lua/world.lua

```diff
@@ -511,7 +534,15 @@ function World:spawnPatient(hospital)
   end
 
   -- Construct patient.
-  local spawn_point = self.spawn_points[math.random(1, #self.spawn_points)]
+  local desks = hospital:findReceptionDesks()
+  local spawns = self.spawn_points
+  local spawn_point = spawns[math.random(1, #spawns)]
+
+  -- Let the first patient to the player use the smallest path to a reception desk
+  if hospital.num_visitors == 0 and class.is(hospital, PlayerHospital) then
+    spawn_point = self:_findClosestSpawnToDesk(desks, spawns)
```

Comment:
So in theory this problem should _never happen_ since we run code at the start of  `World:spawnPatient` to check spawn points and a staffed desk. 
I'd probably like to just make this a gamelog warning with mitigation first now than throw a full on error. In my testing of this even if for some reason a valid path failed the patient still manages to find their way.

Reply:
Gamelog warning added and changed for risk mitigation